src/lib.rs

```diff
@@ -1174,7 +1176,7 @@ impl TestKitApi {
         );
     }
 
-    fn get_internal<H, D>(handler: &H, url: &str, expect_error: bool) -> D
+    fn get_internal<H, D>(handler: &H, endpoint: &str, expect_error: bool, is_public: bool) -> D
```

Comment:
It just an internal method, but I suggest to use struct instead of boolean for better readability:
```rust
struct GetRequest {
    handler: H,
    endpoint: &str,
    expect_error: bool,
    is_public: bool
};
...
api.get_internal(GetRequest { handler, "endpoint", expect_error: true, is_public: false });
```

Reply:
I don't think it's strictly needed, but okay, let it be.

Reply:
Oh well, actually... can we go without it? It just needs too many generics and lifetimes for these structs without much effort.

Reply:
Okay.